frontend/src/components/common/Game/Card/Base.vue

```diff
@@ -110,23 +117,19 @@ const computedAspectRatio = computed(() => {
             @touchend="handleTouchEnd"
             v-bind="hoverProps"
             :class="{ pointer: pointerOnHover }"
-            ref="card"
             cover
             :key="romsStore.isSimpleRom(rom) ? rom.updated_at : ''"
             :src="
-              src
-                ? src
-                : romsStore.isSimpleRom(rom)
-                  ? !rom.igdb_id && !rom.moby_id && !rom.has_cover
-                    ? `/assets/default/cover/big_${theme.global.name.value}_unmatched.png`
-                    : (rom.igdb_id || rom.moby_id) && !rom.has_cover
-                      ? `/assets/default/cover/big_${theme.global.name.value}_missing_cover.png`
-                      : `/assets/romm/resources/${rom.path_cover_l}?ts=${rom.updated_at}`
-                  : !rom.igdb_url_cover && !rom.moby_url_cover
+              src ||
+              (romsStore.isSimpleRom(rom)
+                ? !rom.igdb_id && !rom.moby_id && !rom.has_cover
+                  ? `/assets/default/cover/big_${theme.global.name.value}_unmatched.png`
+                  : (rom.igdb_id || rom.moby_id) && !rom.has_cover
                     ? `/assets/default/cover/big_${theme.global.name.value}_missing_cover.png`
-                    : rom.igdb_url_cover
-                      ? rom.igdb_url_cover
-                      : rom.moby_url_cover
+                    : `/assets/romm/resources/${rom.path_cover_l}?ts=${rom.updated_at}`
+                : !rom.igdb_url_cover && !rom.moby_url_cover
+                  ? `/assets/default/cover/big_${theme.global.name.value}_missing_cover.png`
+                  : rom.igdb_url_cover || rom.moby_url_cover)
```

Comment:
eventually this should be a util function that's more clearly defined/parsable

Reply:
Totally agree, too much logic here

Reply:
Moving this to a separate function in a future PR will also simplify fixing the issue where frontend sometimes sends `?ts=undefined`